src/components/Dialog/index.tsx

```diff
@@ -96,110 +90,100 @@ const Dialog = ({
 
   useEffect(() => {
     if (!alwaysShowBorder) {
-      handleResize(); // needs to fire one time immediately on mount
+      handleResize(); // needs to fire once immediately on mount
```

Comment:
this will also fire on update when `alwaysShowBorder` value changes, if that's what this intended to do then i'm ok with it.

Reply:
I can only assume thats whats wanted because this was here before i worked on it. I only altered the comment here

Reply:
Definitely room for improvements here 😅

Reply:
The thing is that if `alwaysShowBorder` is false, and someone turns it to true, we want the effect to be ran. There's no way in hooks to only run a useEffect conditionally based on the value of a boolean.

However, what you will notice here, is the conditional of `if(!alwaysShowBorder)` which actually means `handleResize` will only run if `alwaysShowBorder` is false.

So in short, the `useEffect` will rerun, but the `handleResize` function will not.